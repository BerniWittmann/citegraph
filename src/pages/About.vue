<template>
  <div class="about">
    <img alt="CiteGraph logo" src="../assets/logo.png">
    <h1>{{ $t('about.title') }}</h1>
    <p>{{ $t('about.text') }}</p>
    <p>{{ $t('about.version', { version: version }) }}</p>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import packageInfo from '@/../package.json'

@Component
export default class AboutPage extends Vue {
  get version (): string {
    return packageInfo.version
  }
}
</script>

<style scoped lang="scss">
.about {
  display: flex;
  flex-direction: column;
  align-items: center;
}
</style>
