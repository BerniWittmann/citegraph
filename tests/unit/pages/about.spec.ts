import { shallowMount } from '@vue/test-utils'
import { i18n } from '../setupPlugins'

import About from '@/pages/About.vue'

describe('pages/About.vue', () => {
  it('renders the page', () => {
    const wrapper = shallowMount(About, {
      i18n
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
