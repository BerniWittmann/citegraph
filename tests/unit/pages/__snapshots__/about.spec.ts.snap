// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/About.vue renders the page 1`] = `
<div class="about"><img alt="CiteGraph logo" src="../assets/logo.png">
  <h1>about.title</h1>
  <p>about.text</p>
  <p>about.version</p>
</div>
`;
